crates/app/src/lib.rs

```diff
@@ -221,6 +221,12 @@ impl AppBuilder {
         self
     }
 
+    #[must_use]
+    pub fn with_shutdown_timeout(mut self, timeout: &str) -> AppBuilder {
```

Comment:
```suggestion
    pub fn with_shutdown_timeout(mut self, timeout: impl Into<String>) -> AppBuilder {
```

nit: If the caller already has an owned string, they can just give us that instead of giving us a reference and us making a copy.